### PR TITLE
feat: Window redefinition.

### PR DESCRIPTION
**Pull request by EdwinBetanc0urt**
## Bug report / Feature

This is a redefinition of the process view and the parent tab component.

#### Steps to reproduce

1. Open path `/test/window/standard`

#### Screenshot or Gif

https://user-images.githubusercontent.com/20288327/123532120-78a08d80-d6d8-11eb-9af0-8aa943ce4e0c.mp4

#### Other relevant information
* Your OS: Linux Mint 19.1 Cinnamon x64. 
* Browser: Mozilla Firefox 88.0.1.
* Node.js version: 14.17.0.
* NPM version: 7.17.0.
* adempiere-vue version: 4.3.1.

#### Additional context
The `RecordNavigation` component is added but not in use, it contains the table and the advanced search panel.

### CODE REVIEW

File: src/components/ADempiere/PanelDefinition/index.vue [hunk 1]
```diff
@@ -20,6 +20,7 @@
   <component
     :is="componentRender"
     :container-uuid="containerUuid"
+    :panel-type="panelType"
```
Comment (yamelsenih) [inline src/components/ADempiere/PanelDefinition/index.vue]: Is unnecessary

File: src/components/ADempiere/PanelDefinition/index.vue [hunk 2]
```diff
@@ -31,13 +32,21 @@ export default defineComponent({
...
+    },
     containerUuid: {
       type: String,
       required: true
     },
     panelType: {
       type: String,
       required: true
+    },
+    isAdvancedQuery: {
```
Comment (yamelsenih) [inline src/components/ADempiere/PanelDefinition/index.vue]: Remove please

File: src/components/ADempiere/PanelDefinition/index.vue [hunk 3]
```diff
@@ -47,37 +56,45 @@ export default defineComponent({
     })
 
     const componentRender = computed(() => {
-      if (['browser', 'process'].includes(props.panelType)) {
```
Comment (yamelsenih) [inline src/components/ADempiere/PanelDefinition/index.vue:50]: Zero hardcode

File: src/components/ADempiere/RecordNavigation/index.vue [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                />
+              </template>
+            </el-collapse-item>
+          </el-collapse>
+        </el-header> -->
+
+        <!-- records in table -->
+        <default-table
+          :parent-uuid="parentUuid"
+          :container-uuid="containerUuid"
```
Comment (yamelsenih) [inline src/components/ADempiere/RecordNavigation/index.vue]: What is the scope of it?

File: src/components/ADempiere/TabParent/index.vue [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+<template>
+  <el-tabs
+    v-model="currentTab"
+    type="border-card"
+    :before-leave="handleBeforeLeave"
+    @tab-click="handleClick"
+  >
+    <template v-for="(tabAttributes, key) in tabsList">
+      <el-tab-pane
```
Comment (yamelsenih) [inline src/components/ADempiere/TabParent/index.vue]: Change by SingleTab

File: src/components/ADempiere/TabParent/index.vue [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+    const setCurrentTab = () => {
+      root.$store.dispatch('setCurrentTab', {
+        parentUuid: props.windowUuid,
+        containerUuid: tabUuid.value,
+        window: props.windowMetadata
+      })
+    }
+
+    const handleBeforeLeave = (activeName) => {
```
Comment (yamelsenih) [inline src/components/ADempiere/TabParent/index.vue]: What is the reason for it? Handle before leave?

**Review by yamelsenih — CHANGES_REQUESTED** [review]
I just need the follow:
- A Window with a single tab
- a store shared between window, tab and other component inside window
- the rest of components are innecesarry

File: src/components/ADempiere/RecordNavigation/index.vue (hunk 4 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/RecordNavigation/index.vue]: Deleted file

File: src/components/ADempiere/TabParent/index.vue (hunk 6 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/TabParent/index.vue]: Delete unused methods

File: src/components/ADempiere/TabParent/index.vue (hunk 5 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/TabParent/index.vue]: Changed file to TabManager

File: src/components/ADempiere/PanelDefinition/index.vue (hunk 2 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/PanelDefinition/index.vue]: Removed this

File: src/components/ADempiere/PanelDefinition/index.vue (hunk 3 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/PanelDefinition/index.vue:50]: This are removed

File: src/components/ADempiere/PanelDefinition/index.vue (hunk 1 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/PanelDefinition/index.vue]: change to panel-metadata prop

**Pull request merged**